src/hooks/useRejoinAsPlayer.ts

```diff
@@ -0,0 +1,46 @@
+import { useAppDispatch, useAppSelector } from "../state/reduxHooks";
+import { useEffect, useState } from "react";
+import { setGameCode, setGameStarted, setHasJoinedGame } from "../state/gameSlice";
+import { resetLeaderboard, setRequestUpdatedLeaderboard } from "../state/leaderboardSlice";
+import Cookies from "js-cookie";
+import { setIsHost, setPlayerName } from "../state/playerSlice";
+import { updateSecondsLeft } from "../state/timeSlice";
+
+export function useRejoinAsPlayer() {
+  const { rejoinAsPlayer } = useAppSelector(state => state.player);
+  const { gameCode } = useAppSelector(state => state.game);
+  const dispatch = useAppDispatch();
+  const [playerRejoinTrigger, setPlayerRejoinTrigger] = useState(false);
+
+  const togglePlayerRejoinTrigger = () => {
+    setPlayerRejoinTrigger(!playerRejoinTrigger);
+  };
+
+  useEffect(() => {
+    const gameCodeCookie = Cookies.get("gameCode");
+    const nameCookie = Cookies.get("name");
+    if (gameCodeCookie !== undefined && nameCookie !== undefined) {
+      dispatch(setIsHost(false));
+      dispatch(setGameCode(gameCodeCookie));
+      dispatch(setPlayerName(nameCookie));
+      dispatch(setGameStarted(true));
+      dispatch(setHasJoinedGame(true));
+      dispatch(resetLeaderboard());
+      dispatch(setRequestUpdatedLeaderboard(false));
```

Comment:
Don't think these two are needed since the player doesn't see the leaderboard

Reply:
gotcha good catch